docs/paper/dev/api/commands.mdx

```diff
@@ -0,0 +1,106 @@
+
+:::
+
+### JavaPlugin
+Commands can be registered inside the <Javadoc name={"org.bukkit.plugin.java.JavaPlugin#onEnable()"}>onEnable</Javadoc>
+method of the plugin. Commands registered here will not be available to datapack functions because
+functions are loaded by the server before plugins are loaded. To make commands available to
+datapacks, register them via the
+<Javadoc name={"io.papermc.paper.plugin.bootstrap.PluginBootstrap#bootstrap(io.papermc.paper.plugin.bootstrap.BootstrapContext)"}>PluginBootstrap</Javadoc>.
+
+```java
+class YourPluginClass extends JavaPlugin {
+
+    @Override
+    public void onEnable() {
+        LifecycleEventManager<Plugin> manager = this.getLifecycleManager();
+        manager.registerEventHandler(LifecycleEvents.COMMANDS, event -> {
+            final Commands commands = event.registrar();
+            commands.register(
+                Commands.literal("new-command")
+                    .executes(ctx -> {
+                        ctx.getSource().getSender().sendPlainMessage("some message");
+                        return Command.SINGLE_SUCCESS;
+                    })
+                    .build(),
+                "some bukkit help description string",
+                List.of("an-alias")
+            );
+        });
+    }
+}
+```
+
+### PluginBootstrap
+
+:::note
+
+If any plugin on the server registers a handler for the `LifecycleEvents.COMMANDS` event, the
+server will disable Bukkit's plugin reload functionality. This is a limitation with how
+the plugin reload system works in conjunction with the plugin bootstrapper system. The `/reload`
+functionality should never be used regardless.
+
+:::
+
+Commands are registered in the same way in a plugin's bootstrapper. The benefit of registering
+commands here is that they will be available to datapack functions because the command registration
+happens early enough.
+
+```java
+class YourPluginBootstrap implements PluginBootstrap {
+
+    @Override
+    public void bootstrap(BootstrapContext context) {
+        LifecycleEventManager<BootstrapContext> manager = context.getLifecycleManager();
+        // the same as JavaPlugin above
+    }
+}
+```
+
+### BasicCommand
+
+Paper provides a simple <Javadoc name={"io.papermc.paper.command.brigadier.BasicCommand"}>BasicCommand</Javadoc>
+interface which can be used to define a command in a similar way to Bukkit's `CommandExecutor`.
+There are methods on the <Javadoc name={"io.papermc.paper.command.brigadier.Commands"}>Commands</Javadoc> interface
+to register such commands.
+
+## Arguments
+
+### Built-in arguments
+
+Vanilla has lots of built-in arguments that the client also supports. These can provide better
+syntax and error-checking before even executing the commands. <Javadoc name={"io.papermc.paper.command.brigadier.argument.ArgumentTypes"}>ArgumentTypes</Javadoc>
+has all the argument types available to the API.
```

Comment:
Maybe it would be useful to have a small example about how to add arguments to a command